### PR TITLE
Fix 7d2d issue when UserDataFolder path contains whitespace

### PR DESCRIPTION
**Pull request by Senjay-id**
Fixes Nexus-Mods/Vortex#16445

### CODE REVIEW

**Comment by Senjay-id** [comment]
Note that it hasn't been tested by me because I don't know how to build it. I'm only adjusting the source file to what was suggested by the issue

File: game-7daystodie/index.tsx [hunk 1]
```diff
@@ -133,7 +133,7 @@ async function prepareForModding(context: types.IExtensionContext,
     }
     await fs.ensureDirWritableAsync(path.join(directory, 'Mods'));
     const launcher = DEFAULT_LAUNCHER_SETTINGS;
-    launcher.DefaultRunConfig.AdditionalParameters = `-UserDataFolder=${directory}`;
+    launcher.DefaultRunConfig.AdditionalParameters = `-UserDataFolder=${directory}"`;
```
Comment (IDCs) [inline game-7daystodie/index.tsx:136]: Thanks @Senjay-id - although please note that you forgot a double quotation mark - the value should've been set to:
`-UserDataFolder="${directory}"`

**Review by IDCs — CHANGES_REQUESTED** [review]
UDF should be set to:
`-UserDataFolder="${directory}"`

**Comment by IDCs** [comment]
Oh btw - bundled extensions are built alongside Vortex; the typescript files are compiled during the build.

File: game-7daystodie/index.tsx (hunk 1 above)
Comment (Senjay-id) [reply game-7daystodie/index.tsx:136]: @IDCs After testing the configuration further, it throws launcher error if we use double quotation mark. I think the only way to fix this is to change how 7d2d parses the configuration in the game's assembly code

`
2024-09-19T17:43:44 0.178 INF Command line arguments: C:\Hub\Steam\steamapps\common\7 Days To Die\7DaysToDie.exe -force-d3d11 -disablenativeinput -nogs -noeac -UserDataFolder=C:\Users\firstname lastname\AppData\Roaming\7daystodie

2024-09-19T17:43:44 0.183 INF UserDataFolder: C:\Users\firstname
`

Comment (Senjay-id) [reply game-7daystodie/index.tsx:136]: I'll take a look at the game assembly code and see if it's possible to fix the parser

Comment (Senjay-id) [reply game-7daystodie/index.tsx]: @IDCs Alright the change should be able to parse udf path that has whitespace, reminder again that I haven't tested the extension but the expected output should be
`
"AdditionalParameters": "-UserDataFolder=\"C:\\Users\\firstname lastname\\AppData\\Roaming\\7daystodie\""
`

File: game-7daystodie/index.tsx [hunk 2]
```diff
@@ -133,7 +133,7 @@ async function prepareForModding(context: types.IExtensionContext,
     }
     await fs.ensureDirWritableAsync(path.join(directory, 'Mods'));
     const launcher = DEFAULT_LAUNCHER_SETTINGS;
-    launcher.DefaultRunConfig.AdditionalParameters = `-UserDataFolder=${directory}`;
+    launcher.DefaultRunConfig.AdditionalParameters = `-UserDataFolder=\"${directory}"`;
```
Comment (IDCs) [inline game-7daystodie/index.tsx:136]: The `\` character is not required as the `JSON.stringify` function will escape double quotations automatically where required.

**Comment by IDCs** [comment]
Thank you for the fix! pushed the compiled files. Keep in mind that you don't have to escape quotation marks when JSON.stringify is called :P

**Review by IDCs — COMMENTED** [review]
Merged and compiled

**Pull request merged**